### ISSUE TITLE
Error: ENOENT, lstat '/Users/josephpea/src/trusktr+/documentjs/site/default/static'

### ISSUE THREAD

**Issue by trusktr** [bitovi/documentjs#95]
I have a folder called `trusktr+documentjs`. From the error it seems like documentjs is adding a slash to it, so that folder doesn't exist?

**Comment by trusktr** [comment]
It's doing it with underscores too: `Error: ENOENT, lstat '/Users/josephpea/src/trusktr_/documentjs/site/default/static'`

**Comment by trusktr** [comment]
By the way, I've got documentjs `npm link`ed, so that's why it's looking there, in the repo I cloned.

**Comment by trusktr** [comment]
Oh duh, it'd help to show you the full error:

```
/Users/josephpea/src/trusktr_documentjs/node_modules/q/q.js:126
                    throw e;
                          ^
Error: ENOENT, lstat '/Users/josephpea/src/trusktr_/documentjs/site/default/static'
```

**Comment by trusktr** [comment]
Maybe it's `q`'s falt?

**Comment by trusktr** [comment]
Even without any symbols `trusktrdocumentjs` results in

```
Error: ENOENT, lstat '/Users/josephpea/src/trusktr/documentjs/site/default/static'
```

**Comment by trusktr** [comment]
just `documentjs` works, so I'll use that temporarily.

**Comment by justinbmeyer** [comment]
Can you explain what your setup looks like in more detail. I don't know what you mean by "I have a folder". What does that mean?

**Comment by trusktr** [comment]
So basically, to install documentjs, I cloned the repo from github, went into the repo folder, then ran `npm link` so that documentjs is installed right from the repo folder globally. Now, when running `documentjs`, I get that error.

DocumentJS is looking for `/Users/josephpea/src/trusktr+/documentjs/site/default/static`, but it can't find it. The actual folder exists, and it is located in a different place: `/Users/josephpea/src/trusktr+documentjs/site/default/static`

DocumentJS (or one of it's dependencies?) seems to be looking in the wrong place.

**Comment by trusktr** [comment]
For comparison:

```sh
/Users/josephpea/src/trusktr+/documentjs/site/default/static # wrong
/Users/josephpea/src/trusktr+documentjs/site/default/static # correct
```

**Comment by justinbmeyer** [comment]
Ok, so you installed documentjs in a folder with a special character.  I will check that out.

Why is that not in a node_modules folder?

**Comment by justinbmeyer** [comment]
ah, global install ... I haven't really tried that yet.

**Comment by justinbmeyer** [comment]
So this line is probably the problem:

https://github.com/bitovi/documentjs/blob/master/lib/fs_extras.js#L11

If you changed that to something like:

```
root = path.dirname(path.dirname(__dirname))
```

Let me know if that fixes it and I will get it into our version.

**Comment by trusktr** [comment]
That didn't work, but `path.dirname(__dirname)` does return the root of my local documentjs, `/Users/josephpea/src/trusktr+documentjs`, so the fix would be not to assume that the app folder is called `documentjs` in whatever other file that is happening in, Documentjs is appending `/documentjs` to whatever the root is.

**Comment by justinbmeyer** [comment]
This is happening in another file?

**Comment by trusktr** [comment]
Yep, the `makdirs`, `copy`, and `exists` functions are receiving paths containing `documentjs`, so the solution seems to be use

```js
root = path.dirname(__dirname)
```

and make sure arguments to those functions don't contain `documentjs`.

**Comment by justinbmeyer** [comment]
Working on this now.  Not sure how this got dropped.  Thanks!

**Comment by trusktr** [comment]
:+1:
